packages/clerk-js/src/ui/common/authPropHelpers.ts

```diff
@@ -78,3 +78,28 @@ export const buildAuthQueryString = (data: BuildAuthQueryStringArgs): string | n
   }
   return Object.keys(query).length === 0 ? null : qs.stringify(query);
 };
+
+export const verifyRoutingOptions = ({
+  routing,
+  path,
+}: {
+  routing?: RoutingStrategy;
+  path?: string;
+}): RoutingOptions => {
+  if (!path) {
+    if (routing === 'path') {
+      throw new Error('Clerk: path routing requires passing the `path` prop');
```

Comment:
🛑 Throwing the error here is great and you had the correct idea :)

Unfortunately, we cannot do this right now. Since we hotload `clerk-js` this is not only a breaking change, but it also has the potential to break live production apps if the user has been using these props incorrectly until now.

Imagine the following scenario:

```
<SignIn routing='hash' path='/sign-in' />
```

Based on the Portal impl [here](https://github.com/clerkinc/javascript/blob/b0d5e73ff718c43382cbf6f2ad23ae9627a8eb9a/packages/clerk-js/src/ui/portal/index.tsx#L1-L0), the above component will use the hash router even if the path prop exists. So in theory there can be a live app using these props in that manner.

I'm adding a [ticket](https://linear.app/clerk/issue/JS-56/throw-an-error-if-routing-component-params-are-used-incorrectly) for now

Reply:
Yeap, I'm aware that this can't throw an error. I was going to remove this once I solve another issue. We can do a `console.error` I guess or nothing at all.

Reply:
Let's do nothing at all for now, we will handle it properly when we work on the ticket I posted above

Reply:
What if `routing='path'` but no `path` prop is passed?